x-pack/platform/plugins/shared/security/server/session_management/session_index.ts

```diff
@@ -475,12 +477,15 @@ export class SessionIndex {
   /**
    * Trigger a removal of any outdated session values.
    */
-  async cleanUp() {
+  async cleanUp(taskManagerRunContext: RunContext) {
+    const { taskInstance } = taskManagerRunContext;
     const { auditLogger, logger } = this.options;
     logger.debug('Running cleanup routine.');
 
     let error: Error | undefined;
     let indexNeedsRefresh = false;
+    let shardMissingCounter = taskInstance.state?.shardMissingCounter ?? 0;
```

Comment:
Not related to this line but would it make sense to reset the counter to `0` whenever the task succeeds? (before the catch block below)